test/e2e/controller_test.go

```diff
@@ -0,0 +1,401 @@
+package e2e
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	kcpclienthelper "github.com/kcp-dev/apimachinery/pkg/client"
+	"github.com/stretchr/testify/assert"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/util/conditions"
+
+	"github.com/kcp-dev/logicalcluster/v2"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	catalogv1alpha1 "github.com/kcp-dev/catalog/api/v1alpha1"
+)
+
+// The tests in this package expect to be called when:
+// - kcp is running
+// - a kind cluster is up and running
+// - it is hosting a syncer, and the SyncTarget is ready to go
+// - the controller-manager from this repo is deployed to kcp
+// - that deployment is synced to the kind cluster
+// - the deployment is rolled out & ready
+//
+// We can then check that the controllers defined here are working as expected.
+
+var workspaceName string
+
+func init() {
+	rand.Seed(time.Now().Unix())
+	flag.StringVar(&workspaceName, "workspace", "", "Workspace in which to run these tests.")
+}
+
+func parentWorkspace(t *testing.T) logicalcluster.Name {
+	flag.Parse()
+	if workspaceName == "" {
+		t.Fatal("--workspace cannot be empty")
+	}
+
+	return logicalcluster.New(workspaceName)
+}
+
+func loadClusterConfig(t *testing.T, clusterName logicalcluster.Name) *rest.Config {
+	t.Helper()
+	restConfig, err := config.GetConfigWithContext("base")
```

Comment:
I'm not sure we want to require a kubeconfig with a context named `base` for these tests to pass? Could we assume the current context in the kubeconfig is sufficient?